src/shared/hooks/use-auto-height.ts

```diff
@@ -8,22 +9,25 @@ interface IConfig {
 export const useAutoHeight = (config: IConfig) => {
   const { container, setHeight } = config;
 
-  const calcHeight = () => {
-    const height = container.current?.offsetHeight;
-    if (height && height > 0) {
-      setHeight(height);
-    }
-  };
-
-  // Update height on every re-render.
-  useEffect(calcHeight);
-
   useEffect(() => {
-    // It's necessary to wrap calcHeight in a closure function. Note that calcHeight will be recreated on every render.
-    // This will break the cleanup function which would try to remove wrong reference. resizeHandler is created
-    // only once, as useEffect has empty dependencies list.
-    const resizeHandler = () => calcHeight();
-    window.addEventListener("resize", resizeHandler);
-    return () => window.removeEventListener("resize", resizeHandler);
-  }, []);
-}
+    // TypeScript doesn't seem to have types of the native ResizeObserver yet. Use types coming from polyfill.
+    const NativeResizeObserver = (window as any).ResizeObserver as new(callback: ResizeObserverCallback) => ResizeObserver;
+
+    const observer = new (NativeResizeObserver || ResizeObserver)(entries => {
```

Comment:
I believe this is unnecessary, as the `ResizeObserver` returned by the module should be the [native ResizeObserver](https://github.com/que-etc/resize-observer-polyfill/blob/4a148452494a155656e4d04b9732b5914896b2e0/src/index.js#L6-L8) if it's available.

Reply:
Nice, I was wondering if that's the case or not. Although the docs suggest that it's a "ponyfill":
https://github.com/sindresorhus/ponyfill
And there's a statement:
> Ponyfills should never use the native API, even if available, as it might have slightly different behavior between environments, which can cause bugs.

So, I thought it's just safer to do the check ourselves anyway.

Reply:
I followed that same thread and concluded that the recommended use is not actually a ponyfill according to Sindre's definition, but it's also not a polyfill since it uses the native implementation if it's present but doesn't patch it in if it's not. I guess we need a new word. Maybe it's a Posyfill, because it poses as the native implementation if it's not present. 😉